app/src/main/java/com/example/datarsd1/pokedex/SearchActivity.java

```diff
@@ -0,0 +1,245 @@
+        inflater.inflate(R.menu.menu_items, menu);
+
+        optionsMenu = menu;
+
+        SearchView pokemonNameSearch = (SearchView) menu.findItem(R.id.name_search).getActionView();
+
+        pokemonNameSearch.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                ArrayList<Pokedex.Pokemon> searchResultList = new ArrayList<Pokedex.Pokemon>();
+                for (int i = 0 ; i < pokemonList.size(); i++)
+                {
+                    if (pokemonList.get(i).name.toLowerCase().startsWith(newText.toLowerCase()) || pokemonList.get(i).number.startsWith(newText))
+                        searchResultList.add(pokemonList.get(i));
+                }
+                pokeadapter.updateList(searchResultList);
+                pokeadapter.notifyDataSetChanged();
+                return true;
+            }
+        });
+        return true;
+    }
+
+    public boolean onOptionsItemSelected(MenuItem item)
+    {
+        switch (item.getItemId()) {
+            case R.id.randomshuffle:
+                ArrayList<Pokedex.Pokemon> randomResultsList = new ArrayList<Pokedex.Pokemon>();
+                for (int i = 0 ; randomResultsList.size() < 20 ; i++)
+                    randomResultsList.add(i,pokemonList.get((int)(Math.random()*pokemonList.size())));
+                pokeadapter.updateList(randomResultsList);
+                pokeadapter.notifyDataSetChanged();
+                break;
+            case R.id.layouttoggle:
+                if (linear) {
+                    searchresults.setLayoutManager(new GridLayoutManager(this, 2));
+                    linear = false;
+                }
+                else {
+                    searchresults.setLayoutManager(new LinearLayoutManager(this));
+                    linear = true;
+                }
+                break;
+        }
+        return true;
+    }
+
+    public void filterbyminstat(double minHP, double minAtk, double minDef)
+    {
+        ArrayList<Pokedex.Pokemon> filterList = new ArrayList<Pokedex.Pokemon>();
+        for (int i = 0 ; i < pokemonList.size(); i++)
+        {
+            if (Double.parseDouble(pokemonList.get(i).hp) > minHP && Double.parseDouble(pokemonList.get(i).attack) > minAtk && Double.parseDouble(pokemonList.get(i).defense) > minDef)
+                filterList.add(pokemonList.get(i));
+        }
+        pokeadapter.updateList(filterList);
+        pokeadapter.notifyDataSetChanged();
+        if (filterList.size() == 0)
+            Toast.makeText(getApplicationContext(), "No results found!", Toast.LENGTH_SHORT).show();
+    }
+
+    public void filterbytype(String type1, String type2)
+    {
+        ArrayList<Pokedex.Pokemon> typeFilterList = new ArrayList<Pokedex.Pokemon>();
```

Comment:
good attempt at separating the filters, but you reset the list in both of them, so filtering by one and then filtering by the other will result in both filters being on but only one being active